src/sentry/tasks/deletion/hybrid_cloud.py

```diff
@@ -57,7 +58,10 @@ def get_watermark(prefix: str, field: HybridCloudForeignKey) -> Tuple[int, str]:
             result = (0, uuid4().hex)
             client.set(key, json.dumps(result))
             return result
-        return tuple(json.loads(v))
+        lower, transaction_id = json.loads(v)
+        if not (isinstance(lower, int) and isinstance(transaction_id, str)):
+            raise TypeError
```

Comment:
this should have a message or it's going to be difficult to track down and/or debug